fiber/test/fiber_tests.ml

```diff
@@ -1112,3 +1112,59 @@ let%expect_test "svar" =
     waiter: waiting for value > 15
     setter: modifying value to 17
     wait: 17 |}]
+
+let%expect_test "map_reduce_seq" =
+  let module Int = struct
+    type t = int
+
+    let zero = 0
+
+    let ( + ) = ( + )
```

Comment:
Addition is not the best monoid for such tests because its commutative, so you won't notice if you reduced elements in the wrong order. I'd suggest using string concatenation instead.

Reply:
I suppose I should mention this in the doc, but there's no "order" for reducing the elements. If one wanted to reduce the sequence in order, there's actually no need for this function and one can write it (almost) equally efficiently without anything added to fiber. Although I suppose that to be complete, we could offer both: `map_reduce_seq_parallel` and `map_reduce_seq_sequential` to be crystal clear about this. Wdyt?

Reply:
Oh, but then the interface is wrong and should ask for a `Commutative_monoid`. Then the result is well-defined regardless of the resulting reduction order.

It's not a sequential vs parallel distinction. You can implement a parallel version that works for any monoid too: just turn need to take care about not reordering the results when they are ready.

Reply:
> Oh, but then the interface is wrong and should ask for a Commutative_monoid. Then the result is well-defined regardless of the resulting reduction order.

I see. We don't currently have a `Cummutative_monoid` in stdune and it would have the same functions as `Monoid`. I  can make a local alias with a comment though`.

> It's not a sequential vs parallel distinction. You can implement a parallel version that works for any monoid too: just turn need to take care about not reordering the results when they are ready.

Yes, but the parallel version for a monoid will need to allocate a large intermediate data structure so it's not all that useful in practice. One could just do `parallel_map` and then reduce manually for approximately the same performance.

Reply:
Do you mind holding off with this PR a bit? I have some `Commutative_monoid` code that I'd be interested in upstreaming to Stdune. Can do this around next week.

Reply:
(Btw, I disagree that a parallel version working with a general `Monoid` is not that useful in practice. It's surely harder to implement it efficiently though, so no need to try until we actually need it.)

Reply:
Sure, we can hold off. There's no rush.

> Commutative_monoid

Perhaps `Monoid.Commutative` would be a bit nicer?

> (Btw, I disagree that a parallel version working with a general Monoid is not that useful in practice. It's surely harder to implement it efficiently though, so no need to try until we actually need it.)

Yes indeed. Although this commutative version will always have a (huge) edge when it's available, so I'm quite eager to  use it.